lms/static/scripts/frontend_apps/utils/url.ts

```diff
@@ -34,10 +34,15 @@ export function replaceURLParams<Param>(
  * Any param which is an array will be appended for every one of its values.
  */
 export function recordToSearchParams(
-  params: Record<string, string | string[]>,
+  params: Record<string, string | string[] | undefined>,
 ): URLSearchParams {
   const queryParams = new URLSearchParams();
   Object.entries(params).forEach(([name, value]) => {
+    // Skip params if their value is undefined
+    if (value === undefined) {
+      return;
+    }
+
```

Comment:
I did `value === undefined` instead of `!value`, because the later will also match for empty strings.

We might want to change this later though. Let's see.

Reply:
I think testing explicitly for `undefined` makes sense. I can imagine a case where we have a boolean flag that is enabled by default and we pass some kind of falsey value (eg. `false`, `0`) to disable it.